packages/hub/node-tests/routes/profiles-test.ts

```diff
@@ -0,0 +1,669 @@
+          color: 'red',
+          textColor: 'purple',
+          ownerAddress: '0x2f58630CA445Ab1a6DE2Bb9892AA2e1d60876C13',
+        },
+      });
+    });
+
+    it('returns 404 if short id cannot be resolved', async function () {
+      await request()
+        .get(`/api/profiles/short-id/now6HxUjmBfDwTbF76GVHz`)
+        .set('Content-Type', 'application/vnd.api+json')
+        .expect(404);
+    });
+
+    it('can return a profile customization', async function () {
+      await request()
+        .get(`/api/profiles/short-id/${shortenedUuid}`)
+        .set('Content-Type', 'application/vnd.api+json')
+        .expect({
+          meta: { network: 'sokol' },
+          data: {
+            id: fullUuid,
+            type: 'profiles',
+            attributes: {
+              did: 'did:cardstack:1mmJKhNVKyAUgScu1dysmR8R728e839d5d105bff',
+              name: 'Merchie!',
+              slug: 'slug',
+              color: 'red',
+              'text-color': 'purple',
+              'owner-address': '0x2f58630CA445Ab1a6DE2Bb9892AA2e1d60876C13',
+              links: [],
+              'profile-description': null,
+              'profile-image-url': null,
+            },
+          },
+        })
+        .expect('Content-Type', 'application/vnd.api+json');
+    });
+  });
+  describe('POST /api/profiles', function () {
+    this.beforeEach(function () {
+      registry(this).register('authentication-utils', StubAuthenticationUtils);
+    });
+
+    let { request } = setupHub(this);
+
+    it('persists profile', async function () {
+      const payload = {
+        data: {
+          type: 'profiles',
+          attributes: {
+            name: 'Satoshi Nakamoto',
+            slug: 'satoshi',
+            color: 'ff0000',
+            'text-color': 'ffffff',
+            'owner-address': '0x00000000000',
+            'profile-description': 'description',
+            'profile-image-url': 'http://example.com/image.jpg',
+          },
+        },
+      };
+
+      await request()
+        .post('/api/profiles')
+        .send(payload)
+        .set('Authorization', 'Bearer abc123--def456--ghi789')
+        .set('Accept', 'application/vnd.api+json')
+        .set('Content-Type', 'application/vnd.api+json')
+        .expect(201)
+        .expect(function (res) {
+          res.body.data.id = 'the-id';
+          res.body.data.attributes.did = 'the-did';
+        })
+        .expect({
+          meta: {
+            network: 'sokol',
+          },
+          data: {
+            type: 'profiles',
+            id: 'the-id',
+            attributes: {
+              name: 'Satoshi Nakamoto',
+              slug: 'satoshi',
+              did: 'the-did',
+              color: 'ff0000',
+              'text-color': 'ffffff',
+              'owner-address': '0x2f58630CA445Ab1a6DE2Bb9892AA2e1d60876C13',
+              links: [],
+              'profile-description': 'description',
+              'profile-image-url': 'http://example.com/image.jpg',
```

Comment:
I wonder if we should prevent storage of these details until we are clearer about what kind of validation/constraints we want to apply?

Reply:
Maybe! Probably? We can make another issue for that if so. These card space properties are near-orphaned 😖